core/templates/services/question-validation.service.ts

```diff
@@ -53,6 +53,34 @@ export class QuestionValidationService {
     );
   }
 
+  hasDuplicateRules(question: Question): boolean {
+    const answerGroups = question._stateData.interaction.answerGroups;
```

Comment:
We should not be directly accessing private members. Is this not obtainable via the public API?

Reply:
> We should not be directly accessing private members. Is this not obtainable via the public API?

@seanlip! You're right about avoiding direct access to private members. In this case, I accessed the private member because there doesn't seem to be a public API to retrieve answerGroups. If there's a public method I should use, I'm happy to refactor accordingly! Let me know your thoughts.